pkg/gce-cloud-provider/compute/gce.go

```diff
@@ -57,13 +59,34 @@ type CloudProvider struct {
 
 var _ GCECompute = &CloudProvider{}
 
-func CreateCloudProvider(vendorVersion string) (*CloudProvider, error) {
-	svc, err := createCloudService(vendorVersion)
+type ConfigFile struct {
+	Global ConfigGlobal `gcfg:"global"`
+}
+
+type ConfigGlobal struct {
+	TokenURL  string `gcfg:"token-url"`
+	TokenBody string `gcfg:"token-body"`
+}
+
+func CreateCloudProvider(vendorVersion string, configPath string) (*CloudProvider, error) {
+	configFile, err := readConfig(configPath)
 	if err != nil {
+		// Error is returned only when configPath is non-empty.
 		return nil, err
 	}
+	glog.Infof("Using GCE provider config %+v", configFile)
 
-	betasvc, err := createBetaCloudService(vendorVersion)
+	tokenSource, err := generateTokenSource(configFile)
```

Comment:
can we do instead `if configFile == nil` then `generateAltToknSource` 
`else` `generateDefaultTokenSource` (or something like thati) instead of doing completely orthogonal logic depending on whether the input is set or not.

Reply:
I'm not sure why that'd be better - isn't this just splitting up the branches one function call level above what's already written?

Reply:
right. the logic ends up being the same but I think its more maintainable.

Instead of having one function do two completely different things based on an input parameter. It makes sense to split those "two different things" into two different functions.

Reply:
I have a slight preference for this version actually, since someone reading this function (`CreateCloudProvider()`) could just skip over the details of how the token source is instantiated if they don't care.

To also have the benefit you mentioned, I could add in a comment in each of the branches to make it explicit they are doing two very different things